### ISSUE TITLE
Backslash required for Dash within square brackets in Regex [.\-_] but Pylint reports it as Anomalous warning

### ISSUE THREAD

**Issue by HurriedPatience** [PyCQA/pylint#3393]
The issue doesn't already exist (open or closed)

### Steps to reproduce
1. Import re

2. re.search(r'(?<=[.\\-_]).*(?=.)', "This is a hyphen-ated word.")
<_sre.SRE_Match object; span=(17, 26), match='ated word'>

3. re.search(r'(?<=[.-_]).*(?=.)', "This is a hyphen-ated word.")
<_sre.SRE_Match object; span=(1, 26), match='his is a hyphen-ated word'>

### Current behavior
Warning "Anomalous backslash in string: '\-'. String constant might be missing an r prefix."

### Expected behavior
Expect no warning because backslash is required within square brackets even with r prefix.  

Without the backslash in square, the meaning of "[.-_]" is any character between the [ascii dot (46)](http://www.asciitable.com/) and [ascii underscore (95)](http://www.asciitable.com/).  Dash is ascii 45.  Matching between is not the intention of that pattern. Matching any one of three characters is the intention.

So this warning could lead someone astray and spend hours testing/trying to figure out something that's working.

Note: For this editor single backslash dash didn't show up in step 2 so I had to put two backslashes dash to make it show in step 2 above.

### pylint --version output
pylint 2.4.4
astroid 2.3.3
Python 3.6.9 (default, Nov  7 2019, 10:44:02) 

You developers are terrific for putting your sweat into Pylint.  Hope this tiny thing helps.

**Comment by asottile** [comment]
I can't reproduce the warning with:

```python3
import re

re.search(r'(?<=[.\-_]).*(?=.)', "This is a hyphen-ated word.")
```

**Comment by PCManticore** [comment]
Hey @HurriedPatience 

Just like @asottile mentioned, I cannot reproduce this as well using:
```
pylint 2.4.4
astroid 2.3.3
Python 3.6.5 (default, May  7 2018, 10:07:07)
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
```

I also tried with the unreleased version:
```
pylint 2.5.0-dev1
astroid 2.4.0
Python 3.6.5 (default, May  7 2018, 10:07:07)
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
```

Let us know if there's something needed to reproduce this. Going to close it for now but feel free to reopen once you can reproduce it as well.

**Comment by HurriedPatience** [comment]
Hi guys,
I missed @asottile's message some how.  

This happens in Visual Code.  (in case this is not assumed)

Also I noticed my python is 3.6.9 while yours is less. (Probably insignificant.)
